packages/analytics-browser-test/test/web-attribution.test.ts

```diff
@@ -0,0 +1,603 @@
+              events: [
+                generateAttributionEvent(++eventId, url),
+                generateSessionStartEvent(++eventId),
+                generatePageViewEvent(++eventId, 1, url),
+                generateSessionEndEvent(++eventId),
+                generateAttributionEvent(++eventId, newCampaignURL),
+                generateSessionStartEvent(++eventId),
+                generateEvent(++eventId, 'test event after session timeout'),
+              ],
+              options: {
+                min_id_length: undefined,
+              },
+            });
+            scope.done();
+            resolve();
+          }, 4000);
+        });
+      });
+
+      test('should not track UTMs and referrers if the referrer is excluded referrer', async () => {
+        const referrer = 'https://www.google.com/';
+        const referring_domain = 'www.google.com';
+        const url = 'https://www.example.com?utm_source=test_utm_source';
+        navigateTo(url, referrer);
+
+        let payload: any = undefined;
+        const scope = nock(httpEndPoint)
+          .post(path, (body: Record<string, any>) => {
+            payload = body;
+            return true;
+          })
+          .reply(200, success);
+
+        await client.init(apiKey, 'user1@amplitude.com', {
+          deviceId: UUID(),
+          defaultTracking: {
+            ...defaultTracking,
+            attribution: {
+              excludeReferrers: [referring_domain],
+            },
+            sessions: true,
+          },
+          sessionTimeout: 500,
+          flushIntervalMillis: 3000,
+        }).promise;
+
+        // update the url and fire the first event after session timeout
+        const newCampaignURL = 'https://www.example.com?utm_source=second_utm_source&utm_content=test_utm_content';
+        setTimeout(() => {
+          navigateTo(newCampaignURL);
+
+          client.track('test event after session timeout');
+        }, 600);
+
+        return new Promise<void>((resolve) => {
+          setTimeout(() => {
+            expect(payload).toEqual({
+              api_key: apiKey,
+              client_upload_time: event_upload_time,
+              events: [
+                generateSessionStartEvent(++eventId),
+                generatePageViewEvent(++eventId, 1, url, referrer),
+                generateSessionEndEvent(++eventId),
+                generateSessionStartEvent(++eventId),
+                generateEvent(++eventId, 'test event after session timeout'),
+              ],
+              options: {
+                min_id_length: undefined,
+              },
+            });
+            scope.done();
+            resolve();
+          }, 4000);
+        });
+      });
+    });
+
+    describe('during a session', () => {
+      let eventId = -1;
+      beforeEach(() => {
+        eventId = -1;
+      });
+
+      afterEach(() => {
+        cleanup();
+      });
+
+      test('should not track updated campaign during', async () => {
```

Comment:
Worth adding a comment here that this is for a SPA. Because when the page is reloaded, the init will track a campaign change